v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/convertors/ChangeEventSessionConvertor.java

```diff
@@ -169,6 +177,35 @@ JsonNode removeDroppedColumns(JsonNode changeEvent, String tableId) {
     return changeEvent;
   }
 
+  public JsonNode transformChangeEventViaOverrides(JsonNode changeEvent)
+      throws InvalidChangeEventException {
+    String sourceTableName = changeEvent.get(EVENT_TABLE_NAME_KEY).asText();
+    String spTableName = schemaOverridesParser.getTableOverrideOrDefault(sourceTableName);
+    // Replace the source table name with the overridden spanner table name if the override
+    // is specified at the table level.
+    if (!sourceTableName.equals(spTableName)) {
+      ((ObjectNode) changeEvent).put(EVENT_TABLE_NAME_KEY, spTableName);
+    }
+    // Get the list of sourceColumnNames from the event
+    List<String> sourceFieldNames = ChangeEventUtils.getEventColumnKeys(changeEvent);
+    sourceFieldNames.forEach(
+        sourceFieldName -> {
+          Pair<String, String> spannerTableColumn =
+              schemaOverridesParser.getColumnOverrideOrDefault(sourceTableName, sourceFieldName);
+          // a valid column override for the table in this changeEvent exist
+          // 1.  the table name of the source should match the one specified in the override
+          // 2. the column name override should be a different value than the current source field
+          // name.
+          if (sourceTableName.equals(spannerTableColumn.getLeft())
```

Comment:
is this check needed ? the source table is being passed to the override parser in any case

Reply:
Agreed that this check is not needed, I just wanted to be more explicit and implement the two conditions written above in the docs.

Reply:
This seems quite confusing which stems from the return type pair of the interface being confusing.

Case in point. The check here seems to verify sourceTableName against the SpannerTableColumn.getLeft() which seems odd - what is being read the source or spanner ?

It took me a while to figure out which table is being returned. even the interface javadocs don't mention this. I suggest just keep the interface simple and return the column name for given source table. Let the implementation of the interface ensure that the table is retreived correctly or default value is handled. This seems like a leak in responsibility.

Reply:
Done